### ISSUE TITLE
Websockets 14 does not accept Transfer-Encoding header in error responses

### ISSUE THREAD

**Issue by Jolmberg** [python-websockets/websockets#1550]
After upgrading to websockets 14, websockets.connect is raising the exception `NotImplementedError: transfer codings aren't supported` in a scenario that previously raised InvalidStatusCode (503). I was expecting it to change to InvalidStatus, which would have been fine, but even if I catch the NotImplementedError, there is no way to extract an error code from it. The issue arises when my server, in response to the websocket upgrade request returns a 503 error, with `transfer-encoding: chunked` and then a short error message in the body. While it could be argued that this particular message could just use Content-Length rather than chunked encoding, this does seem like a regression on the part of websockets.

The entire back and forth looks something like this.
Request (sent with websockets 14.1):
```
GET /path HTTP/1.1
Host: 192.168.0.1:8888
Upgrade: websocket
Connection: Upgrade
Sec-WebSocket-Key: bDKBNWhBF+86IyQBcT176A==
Sec-WebSocket-Version: 13
Sec-WebSocket-Extensions: permessage-deflate; client_max_window_bits
Sec-WebSocket-Protocol: xyz
Authorization: Bearer 97eab12b-2766-46ca-9ec4-f65d8d26d64c
User-Agent: Python/3.12 websockets/14.1
```
Response:
```
HTTP/1.1 503 Service Unavailable
date: Fri, 15 Nov 2024 08:17:57 GMT
server: Cowboy
transfer-encoding: chunked

This did not go well
```

**Comment by aaugustin** [comment]
This is happening because the legacy implementation didn't even attempt to read the HTTP response body. I added that functionality in the standard case but not in the chunked encoding case.

I believe that #1549 will wrap that error in an `InvalidMessage`, which is closer to what you want. However, it still doesn't give you the HTTP error code, because websockets parses the entire request first and only then starts the opening handshake logic.

I could make an argument for skipping the body rather than crashing in this case. Let me think about it.

**Comment by aaugustin** [comment]
I forgot to mention — as a short term fix, you can use `websockets.legacy.client.connect` instead of `websockets.connect` and you buy five years. See https://websockets.readthedocs.io/en/latest/project/changelog.html and https://websockets.readthedocs.io/en/latest/howto/upgrade.html for context.

**Comment by aaugustin** [comment]
Yes, it's a legitimate request for improvement. The HTTP code is right there and it's reasonable to expect websockets to give it to you.

**Comment by aaugustin** [comment]
I took a stab at fixing it in #1568.

I wanted to test against a real life server rather than just my test suite but I couldn't find one in 15 minutes and I'm not really excited with the idea of installing Erlang and Cowboy locally (-:

Any chance you can test the branch before I merge the PR?

**Comment by Jolmberg** [comment]
Thanks for looking into it, I can confirm that the issue-1550 branch fixes the problem on my end!

**Comment by aaugustin** [comment]
Great, thank you for reporting this issue and testing the fix!
